File: leetcode/PlusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        carry = 1
        i = len(digits) - 1
        
        while i > -1 and carry:
            digits[i] += carry
            if digits[i] > 9:
                digits[i] = 0
            else:
                carry = 0
                
            i -= 1
            
        if i <= 0 and carry:
            digits.insert(0, 1)
        
        return digits
        

File: leetcode/PlusOne.js
var plusOne = function(array) {
    for (let i = array.length - 1; i >= 0; i--) {
    if (array[i] === 9) {
        array[i] = 0
    } else {
        array[i] = array[i] + 1
        return array
        }
    }

    array.unshift(1)
    return array
};
